feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -425,104 +387,109 @@ fun DetectWindowColorModeChanges(
  */
 @Composable
 fun PreviewDisplay(
-    previewUiState: PreviewUiState.Ready,
+    previewDisplayUiState: PreviewDisplayUiState,
     onTapToFocus: (x: Float, y: Float) -> Unit,
     onFlipCamera: () -> Unit,
     onZoomRatioChange: (CameraZoomRatio) -> Unit,
     onRequestWindowColorMode: (Int) -> Unit,
-    aspectRatio: AspectRatio,
     surfaceRequest: SurfaceRequest?,
     modifier: Modifier = Modifier
 ) {
-    val transformableState = rememberTransformableState(
-        onTransformation = { pinchZoomChange, _, _ ->
-            onZoomRatioChange(
-                CameraZoomRatio(
-                    ZoomChange.Scale(pinchZoomChange)
+    if (previewDisplayUiState.aspectRatioUiState is AspectRatioUiState.Available) {
```

Comment:
nit: instead of nesting all of the logic here inside `if (previewDisplayUiState.aspectRatioUiState is AspectRatioUiState.Available) { }`, you could just add

```
if (previewDisplayUiState.aspectRatioUiState !is AspectRatioUiState.Available) {
    return
}
```

to the top of the function.